### PR TITLE
qcnn docs

### PR DESCRIPTION
**Pull request by arthurfaria**
Hello guys, I completed the docs for the qcnn model, addressing finally the [issue](https://github.com/pasqal-io/qadence/issues/727). Please let me know your comments and thoughts. Thanks!

### CODE REVIEW

File: docs/tutorials/qml/qcnn.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+$$\text{C}_l^{\bm{\theta}_l}(\cdot) = \bigcirc_{j=1}^{r}  \left(\bigotimes_{i \in \text{S}(j)} W_l^{(i, i+1)}\left(\bm{\theta}_l\right)\right)(\cdot)$$
+
+Each convolutional layer acts on an even number of qubits, since the unitary $W_l$\footnote{We also refer to $W$ as convolutional cell.} convolves a pairs of neighboring qubits $(i, i+1)$.  In general, the operator $W$ acts on pairs of adjacent qubits defined by the set $S(j) = \{(i, i+1) \mid i \equiv j \!\!\!\mod 2, \ 0 \leq i \leq N-2\}$, where $N$ denotes the total number of qubits\footnote{Qubit indexing is 0-based, i.e., $0, 1, \dots, N-1$.}. This construction ensures an alternating nearest-neighbor interaction pattern. The alternation is determined by the parity of $j$: for even $j$, $W$ acts on even-indexed pairs $(0,1)$, $(2,3)$, etc.; similarly, for odd values of $j$, the same operation is performed on odd-indexed pairs in an analogous manner. Each convolutional layer $\text{C}_l^{\bm{\theta}_l}$ has an associated parameter $r$, representing its depth. For example, in a two-layer QGCN architecture, the depths of $\text{C}_1$ and $\text{C}_2$ are denoted as $r_1$ and $r_2$, respectively.
```
Comment (sungwoo-pasqal) [inline docs/tutorials/qml/qcnn.md:15]: `footnote` in latex is not working well in markdown format.

File: docs/tutorials/qml/qcnn.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+$$\text{P}_{l}^{\bm{\lambda}_l}(\cdot) = Tr_{i}[(\cdot)],$$
+
+where the $i$-qubit is traced out in the $l$-th layer. The pooling layers typically discard half of the qubits at each step.
+
+We adopt a simple architecture for $\text{C}_l^{\bm{\theta}_l}$ and $\text{P}_{l}^{\bm{\lambda}_l}$. The unitary $W$ is defined as in Ref. [vatan_2004_optimal](https://arxiv.org/abs/quant-ph/0308006), where the $A$ gates are defined similarly in terms of $R_G(\bm{\theta}_l) = e^{-iX\theta^1_{l}/2}e^{-iZ\theta^2_{l}/2}e^{-iX\theta^3_{l}/2}$. Entanglement is achieved applying non-parametrized CZ gates. In the pooling layers, entanglement is followed by local measurements on $\mathcal{O}$, enabled by the deferred measurement principle as in Ref. [Nielsen](https://books.google.com.br/books?hl=en&lr=&id=-s4DEy7o-a0C&oi=fnd&pg=PR17&ots=NJ5JfksuWw&sig=Un_kdl0BZ-eJzwRuS2JuoGY1KYI&redir_esc=y#v=onepage&q&f=false). The chosen design reduces the complexity of the QGCN circuit~\eqref{eq:qgcn-map} by
```
Comment (sungwoo-pasqal) [inline docs/tutorials/qml/qcnn.md:22]: Let’s follow the same reference format as in the other tutorial files.

Comment (sungwoo-pasqal) [inline docs/tutorials/qml/qcnn.md:22]: `eqref` also need to be replaced.

File: docs/tutorials/qml/qcnn.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+1. 4 grid elements (each with `n_qubits` features) influence a 5th element
+2. The relationship is encoded in a single continuous target value
+
+#### Input and Target Tensors (`dummy_input` and `dummy_target`)
+1. Input
+    - Shape: `(4, n_qubits)`: 4 samples representing 4 grid elements with `n_qubits` features per grid element. Random values between [0, 1)
+2. Target
+    - Shape: `(1, 1)`: Single scalar value representing the aggregated property of the 5th connected grid element. Random value between [0, 1)
+
+```python
```
Comment (sungwoo-pasqal) [inline docs/tutorials/qml/qcnn.md:39]: It would be good to follow the same code formatting style as in the other document files.

File: docs/tutorials/qml/qcnn.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+```
+
+## Training a QCNN
+
+Now we perform the training of the QCNN on the dummy data generated earlier.
+
+### Define the QCNN circuit with Qadence
+First we define a class with mean pooling on the QCNN output to match target value
+```
+class qcnn_msg_passing(torch.nn.Module):
```
Comment (sungwoo-pasqal) [inline docs/tutorials/qml/qcnn.md:53]: Same code formatting here and below.

File: docs/tutorials/qml/qcnn.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+for epoch in range(n_epochs + 1):
+    optimizer.zero_grad()
+    output = model(dummy_input)
+    loss = torch.nn.functional.mse_loss(output, dummy_target)
+    loss.backward()
+    optimizer.step()
+
+    if epoch % 10 == 0:
+        print(f"Epoch={epoch:>4d} | Loss={loss.item():>10.6f}")
+```
```
Comment (sungwoo-pasqal) [inline docs/tutorials/qml/qcnn.md:107]: It would be better to show some results or figures. If you can give the pdf or plot file, I'll add it.

**Review by sungwoo-pasqal — COMMENTED** [review]
Some comments regarding the format. I'll try to refine most of them. Please share the file with the result that you mentioned with me.

File: docs/tutorials/qml/qcnn.md (hunk 5 above)
Comment (arthurfaria) [reply docs/tutorials/qml/qcnn.md:109]: A final comment: equations are not being displayed right. Try to have the equations in $$ $$, it might help

Comment (sungwoo-pasqal) [reply docs/tutorials/qml/qcnn.md:109]: Oh it is showing right at mkdoc in html. It might be different in current preview.

**Review by sungwoo-pasqal — APPROVED** [review]
